src/snowflake/cli/api/entities/utils.py

```diff
@@ -306,7 +307,10 @@ def render_script_templates(
         script_full_path = SecurePath(project_root) / relpath
         try:
             template_content = script_full_path.read_text(file_size_limit_mb=UNLIMITED)
-            result = render_from_str(template_content, jinja_context)
+            env = override_env or choose_sql_jinja_env_based_on_template_syntax(
+                template_content, reference_name=relpath
+            )
+            result = env.from_string(template_content).render(jinja_context)
```

Comment:
Would it make sense to always pass env in to `render_script_template` instead of having it as an optional override?

Reply:
It doesn't work because each file might have different env (based on what syntax is used in each file)